### PR TITLE
Added fluorite block

### PR DESCRIPTION
**Pull request by kingdevnl**
## Changes proposed in this pull request:
Added a fluorite block

This can close https://github.com/mekanism/Mekanism-Feature-Requests/issues/297

### CODE REVIEW

**Comment by kingdevnl** [comment]
The only thing left is a texture for the block, Iv'e just made a simple texture for it (I'm not a texture artist)

File: src/datagen/main/java/mekanism/common/tag/MekanismTagProvider.java [hunk 1]
```diff
@@ -297,14 +297,15 @@ private void addOres() {
...
         addToTags(MekanismTags.Items.STORAGE_BLOCKS_STEEL, MekanismTags.Blocks.STORAGE_BLOCKS_STEEL, MekanismBlocks.STEEL_BLOCK);
         getItemBuilder(Tags.Items.STORAGE_BLOCKS).add(MekanismTags.Items.STORAGE_BLOCKS_BRONZE, MekanismTags.Items.STORAGE_BLOCKS_CHARCOAL,
-              MekanismTags.Items.STORAGE_BLOCKS_REFINED_GLOWSTONE, MekanismTags.Items.STORAGE_BLOCKS_REFINED_OBSIDIAN, MekanismTags.Items.STORAGE_BLOCKS_STEEL);
+                MekanismTags.Items.STORAGE_BLOCKS_REFINED_GLOWSTONE, MekanismTags.Items.STORAGE_BLOCKS_REFINED_OBSIDIAN, MekanismTags.Items.STORAGE_BLOCKS_STEEL);
         getBlockBuilder(Tags.Blocks.STORAGE_BLOCKS).add(MekanismTags.Blocks.STORAGE_BLOCKS_BRONZE, MekanismTags.Blocks.STORAGE_BLOCKS_CHARCOAL,
-              MekanismTags.Blocks.STORAGE_BLOCKS_REFINED_GLOWSTONE, MekanismTags.Blocks.STORAGE_BLOCKS_REFINED_OBSIDIAN, MekanismTags.Blocks.STORAGE_BLOCKS_STEEL);
+                MekanismTags.Blocks.STORAGE_BLOCKS_REFINED_GLOWSTONE, MekanismTags.Blocks.STORAGE_BLOCKS_REFINED_OBSIDIAN, MekanismTags.Blocks.STORAGE_BLOCKS_STEEL);
```
Comment (pupnewfster) [inline src/datagen/main/java/mekanism/common/tag/MekanismTagProvider.java:308]: These two tags should have the STORAGE_BLOCKS_FLUORITE tag be added to them.

File: src/datagen/main/java/mekanism/common/tag/MekanismTagProvider.java [hunk 2]
```diff
@@ -416,4 +417,4 @@ private void addInfuseTags() {
         addToTag(MekanismTags.InfuseTypes.FUNGI, MekanismInfuseTypes.FUNGI);
         addToTag(MekanismTags.InfuseTypes.BIO, MekanismInfuseTypes.BIO);
     }
-}
\ No newline at end of file
+}
```
Comment (pupnewfster) [inline src/datagen/main/java/mekanism/common/tag/MekanismTagProvider.java]: Is there a reason you are changing whether or not there is a new line at end of the different java files?

**Review by pupnewfster — COMMENTED** [review]
One thing I am a bit hesitant on is the texture looks really out of place for Mekanism, though I will talk some with our texture artists to see if they have any interest in creating a texture for the block.

File: src/datagen/main/java/mekanism/common/tag/MekanismTagProvider.java (hunk 2 above)
Comment (kingdevnl) [reply src/datagen/main/java/mekanism/common/tag/MekanismTagProvider.java]: I think my IDE must of have done somehow

File: src/datagen/main/java/mekanism/common/tag/MekanismTagProvider.java (hunk 1 above)
Comment (pupnewfster) [reply src/datagen/main/java/mekanism/common/tag/MekanismTagProvider.java]: I think you forgot to rerun the datagen after adding them now

**Comment by pupnewfster** [comment]
The main things missing currently I think (given our texture artists are willing to make a texture that fits Mekanism a bit better) is:
- Making this target the 10.1 branch and not the 1.16.x branch (so that I don't have to then merge it into the 10.1 branch at some point given it is a bit of a pain with how many things have changed including mappings)
- Rerunning the datagen. Both so that it adds the fluorite storage blocks to the general storage blocks tags, and so that once it is on the 10.1 branch the `.cache` file actually updates
- Maybe add a recipe like [this](https://github.com/mekanism/Mekanism/blob/v10.1/src/datagen/main/java/mekanism/common/recipe/impl/OreProcessingRecipeProvider.java#L446-L451) that converts the block, 9 mB of sulfuric acid into 9,000 mB of hydrofluoric acid. Not sure if we should add one or not but there probably isn't any harm in doing so (the recipe could probably just be `hydrofluoric_acid_from_block`).

**Comment by kingdevnl** [comment]
I will work on that this weekend.

**Pull request closed without merging**